### PR TITLE
api: fix namespace return errors

### PR DESCRIPTION
**Pull request by eduardoveiga**

### CODE REVIEW

**Review by noreng-jg — DISMISSED** [review]

**Review by otavio — CHANGES_REQUESTED** [review]
The change looks fine. However, I'd also to have a test to avoid this kind of error return. Can you add it?

File: api/nsadm/service.go [hunk 1]
```diff
@@ -45,6 +45,32 @@ func NewService(store store.Store) Service {
 	return &service{store}
 }
 
+func (s *service) isNamespaceOwner(ctx context.Context, tenantID, ownerID string) (*models.Namespace, error) {
```
Comment (otavio) [inline api/nsadm/service.go:48]: GetNamespaceForOwner

File: api/nsadm/service.go [hunk 2]
```diff
@@ -219,7 +198,15 @@ func (s *service) RemoveNamespaceUser(ctx context.Context, tenantID, username, o
 }
 
 func (s *service) EditSessionRecordStatus(ctx context.Context, sessionRecord bool, tenant, ownerID string) error {
-	if _, err := s.GetNamespaceForOwner(ctx, tenant, ownerID); err != nil {
+	err := utils.IsNamespaceOwner(ctx, s.store, tenant, ownerID)
+	if err != nil {
```
Comment (otavio) [inline api/nsadm/service.go:202]: This can be checked on a single line; there are other similar cases too.

**Review by otavio — CHANGES_REQUESTED** [review]

File: api/routes/nsadm.go [hunk 3]
```diff
@@ -142,6 +144,9 @@ func EditNamespace(c apicontext.Context) error {
 
 	namespace, err := svc.EditNamespace(c.Ctx(), c.Param("id"), req.Name, id)
 	if err != nil {
+		if err == nsadm.ErrInvalidFormat {
+			return c.NoContent(http.StatusBadRequest)
+		}
```
Comment (otavio) [inline api/routes/nsadm.go:149]: This is a perfect usecase for the switch.

**Review by otavio — DISMISSED** [review]

**Review by otavio — APPROVED** [review]

**Review by noreng-jg — APPROVED** [review]

**Pull request merged**